scripts/tpdecoder.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+from hdf5libs import HDF5RawDataFile
+import h5py
+
+import daqdataformats
+import trgdataformats
+
+import click
+import time
+import numpy as np
+
+from rawdatautils.unpack.dataclasses import *
+
+@click.command()
+@click.argument('filenames', nargs=-1, type=click.Path(exists=True))
+@click.option('--nrecords', '-n', default=-1, help='How many Trigger Records to process (default: all)')
+@click.option('--nskip', default=0, help='How many Trigger Records to skip (default: 0)')
+
+def main(filenames, nrecords, nskip):
+
+    for filename in filenames:
+
+        
+        h5_file = HDF5RawDataFile(filename)
+        records = h5_file.get_all_record_ids()
+        
+        if nskip > len(records):
+            print(f'Requested records to skip {nskip} is greater than number of records {len(records)}. Exiting...')
+            return
+        if nrecords>0:
+            if (nskip+nrecords)>len(records):
+                nrecords=-1
+            else:
+                nrecords=nskip+nrecords
+
+        records_to_process = []
+        if nrecords==-1:
+            records_to_process = records[nskip:]
+        else:
+            records_to_process = records[nskip:nrecords]
+        print(f'Will process {len(records_to_process)} of {len(records)} records.')
+
+        
+        with h5py.File(h5_file.get_file_name(), 'r') as f:
+            record_type = f.attrs["record_type"]
```

Comment:
Was this intended for something?

Reply:
It's not used, but was intended to be able to extract if it's a trigger record or timeslice record. I've left it in, but adding a print statement to inform.